Rust build errors

See the below picture （System: Windows11, use openblas installed from vcpkg）

<img width="3020" height="2034" alt="Image" src="https://github.com/user-attachments/assets/25c9dd90-7c15-44b3-9d54-fc6edb9d4ed7" />

I try to use --no-default-features parameter to build again today (with v0.6.2 source code), it still shows errors, see the below

```
Compiling sysinfo v0.32.1
warning: `wifi-densepose-signal` (lib) generated 5 warnings (run `cargo fix --lib -p wifi-densepose-signal` to apply 1 suggestion)
error[E0432]: unresolved import `ruvector_mincut`
 --> crates\wifi-densepose-sensing-server\src\csi.rs:5:5
  |
5 | use ruvector_mincut::{DynamicMinCut, MinCutBuilder};
  |     ^^^^^^^^^^^^^^^ use of unresolved module or unlinked crate `ruvector_mincut`
  |
  = help: if you wanted to use a crate named `ruvector_mincut`, use `cargo add ruvector_mincut` to add it to your `Cargo.toml`

error[E0432]: unresolved import `ruvector_mincut`
  --> crates\wifi-densepose-sensing-server\src\main.rs:27:5
   |
27 | use ruvector_mincut::{DynamicMinCut, MinCutBuilder};
   |     ^^^^^^^^^^^^^^^ use of unresolved module or unlinked crate `ruvector_mincut`
   |
   = help: if you wanted to use a crate named `ruvector_mincut`, use `cargo add ruvector_mincut` to add it to your `Cargo.toml`

error[E0425]: cannot find value `CLASSES` in module `adaptive_classifier`
    --> crates\wifi-densepose-sensing-server\src\main.rs:3227:45
     |
3227 |             "classes": adaptive_classifier::CLASSES,
     |                                             ^^^^^^^ not found in `adaptive_classifier`

warning: unused imports: `KeypointState`, `PoseTrack`, and `self`
  --> crates\wifi-densepose-sensing-server\src\tracker_bridge.rs:10:5
   |
10 |     self, KeypointState, PoseTrack, TrackLifecycleState, TrackId, NUM_KEYPOINTS,
   |     ^^^^  ^^^^^^^^^^^^^  ^^^^^^^^^
   |
   = note: `#[warn(unused_imports)]` (part of `#[warn(unused)]`) on by default

error[E0499]: cannot borrow `s` as mutable more than once at a time
    --> crates\wifi-densepose-sensing-server\src\main.rs:1662:39
     |
1661 |         let tracked = tracker_bridge::tracker_update(
     |                       ------------------------------ first borrow later used by call
1662 |             &mut s.pose_tracker, &mut s.last_tracker_instant, raw_persons,
     |                  -                    ^ second mutable borrow occurs here
     |                  |
     |                  first mutable borrow occurs here

error[E0499]: cannot borrow `s` as mutable more than once at a time
    --> crates\wifi-densepose-sensing-server\src\main.rs:1798:35
     |
1797 |     let tracked = tracker_bridge::tracker_update(
     |                   ------------------------------ first borrow later used by call
1798 |         &mut s.pose_tracker, &mut s.last_tracker_instant, raw_persons,
     |              -                    ^ second mutable borrow occurs here
     |              |
     |              first mutable borrow occurs here

error[E0502]: cannot borrow `s` as immutable because it is also borrowed as mutable
    --> crates\wifi-densepose-sensing-server\src\main.rs:3614:43
     |
3613 |                     if let Some(ref mut fm) = s.field_model {
     |                                               - mutable borrow occurs here
3614 |                         if let Some(ns) = s.node_states.get(&node_id) {
     |                                           ^ immutable borrow occurs here
3615 |                             field_bridge::maybe_feed_calibration(fm, &ns.frame_history);
     |                                                                  -- mutable borrow later used here

error[E0499]: cannot borrow `s` as mutable more than once at a time
    --> crates\wifi-densepose-sensing-server\src\main.rs:3699:51
     |
3698 |                     let tracked = tracker_bridge::tracker_update(
     |                                   ------------------------------ first borrow later used by call
3699 |                         &mut s.pose_tracker, &mut s.last_tracker_instant, raw_persons,
     |                              -                    ^ second mutable borrow occurs here
     |                              |
     |                              first mutable borrow occurs here

error[E0502]: cannot borrow `s` as immutable because it is also borrowed as mutable
    --> crates\wifi-densepose-sensing-server\src\main.rs:3862:43
     |
3861 |                     if let Some(ref mut fm) = s.field_model {
     |                                               - mutable borrow occurs here
3862 |                         if let Some(ns) = s.node_states.get(&node_id) {
     |                                           ^ immutable borrow occurs here
3863 |                             field_bridge::maybe_feed_calibration(fm, &ns.frame_history);
     |                                                                  -- mutable borrow later used here

error[E0499]: cannot borrow `s` as mutable more than once at a time
    --> crates\wifi-densepose-sensing-server\src\main.rs:3909:51
     |
3908 |                     let tracked = tracker_bridge::tracker_update(
     |                                   ------------------------------ first borrow later used by call
3909 |                         &mut s.pose_tracker, &mut s.last_tracker_instant, raw_persons,
     |                              -                    ^ second mutable borrow occurs here
     |                              |
     |                              first mutable borrow occurs here

error[E0499]: cannot borrow `s` as mutable more than once at a time
    --> crates\wifi-densepose-sensing-server\src\main.rs:4045:39
     |
4044 |         let tracked = tracker_bridge::tracker_update(
     |                       ------------------------------ first borrow later used by call
4045 |             &mut s.pose_tracker, &mut s.last_tracker_instant, raw_persons,
     |                  -                    ^ second mutable borrow occurs here
     |                  |
     |                  first mutable borrow occurs here

Some errors have detailed explanations: E0425, E0432, E0499, E0502.
For more information about an error, try `rustc --explain E0425`.
warning: `wifi-densepose-sensing-server` (bin "sensing-server") generated 1 warning
error: could not compile `wifi-densepose-sensing-server` (bin "sensing-server") due to 10 previous errors; 1 warning emitted
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -7,6 +7,21 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+### Fixed
+- **Rust workspace build with `--no-default-features` on Windows** (#366, #415) —
+  `wifi-densepose-mat`, `wifi-densepose-sensing-server`, and `wifi-densepose-train`
+  all depended on `wifi-densepose-signal` with default features enabled, which
+  pulled `ndarray-linalg` → `openblas-src` → vcpkg/system-BLAS through the entire
+  workspace. `--no-default-features` at the workspace root then could not opt out
+  of BLAS, breaking `cargo build` / `cargo test` on Windows without vcpkg. All
+  three consumers now declare `wifi-densepose-signal = { ..., default-features = false }`,
+  so `cargo test --workspace --no-default-features` builds cleanly without
+  vcpkg/openblas. Validated: 1,538 tests pass, 0 fail, 8 ignored.
+- **`signal` test `test_estimate_occupancy_noise_only` failed without `eigenvalue`** —
+  The test unwrapped the `NotCalibrated` stub returned when the BLAS-backed
+  `estimate_occupancy` is compiled out. Gated with `#[cfg(feature = "eigenvalue")]`
+  so it only runs when the real implementation is available.
+
 ## [v0.6.2-esp32] — 2026-04-20
 
 Firmware release cutting ADR-081 and the Timer Svc stack fix discovered during

--- a/rust-port/wifi-densepose-rs/crates/wifi-densepose-mat/Cargo.toml
+++ b/rust-port/wifi-densepose-rs/crates/wifi-densepose-mat/Cargo.toml
@@ -25,7 +25,7 @@ serde = ["dep:serde", "chrono/serde", "geo/use-serde"]
 [dependencies]
 # Workspace dependencies
 wifi-densepose-core = { version = "0.3.0", path = "../wifi-densepose-core" }
-wifi-densepose-signal = { version = "0.3.0", path = "../wifi-densepose-signal" }
+wifi-densepose-signal = { version = "0.3.0", path = "../wifi-densepose-signal", default-features = false }
 wifi-densepose-nn = { version = "0.3.0", path = "../wifi-densepose-nn" }
 ruvector-solver = { workspace = true, optional = true }
 ruvector-temporal-tensor = { workspace = true, optional = true }

--- a/rust-port/wifi-densepose-rs/crates/wifi-densepose-sensing-server/Cargo.toml
+++ b/rust-port/wifi-densepose-rs/crates/wifi-densepose-sensing-server/Cargo.toml
@@ -44,8 +44,11 @@ clap = { workspace = true }
 # Multi-BSSID WiFi scanning pipeline (ADR-022 Phase 3)
 wifi-densepose-wifiscan = { version = "0.3.0", path = "../wifi-densepose-wifiscan" }
 
-# Signal processing with RuvSense pose tracker (accuracy sprint)
-wifi-densepose-signal = { version = "0.3.0", path = "../wifi-densepose-signal" }
+# Signal processing with RuvSense pose tracker (accuracy sprint).
+# default-features = false drops the optional ndarray-linalg/BLAS chain so that
+# `--no-default-features` at the workspace root can produce a Windows-friendly
+# build without vcpkg/openblas (issue #366, #415).
+wifi-densepose-signal = { version = "0.3.0", path = "../wifi-densepose-signal", default-features = false }
 
 [dev-dependencies]
 tempfile = "3.10"

--- a/rust-port/wifi-densepose-rs/crates/wifi-densepose-signal/src/ruvsense/field_model.rs
+++ b/rust-port/wifi-densepose-rs/crates/wifi-densepose-signal/src/ruvsense/field_model.rs
@@ -1232,6 +1232,9 @@ mod tests {
         }
     }
 
+    // estimate_occupancy() falls back to a NotCalibrated stub without the
+    // `eigenvalue` feature, so this test only makes sense with BLAS enabled.
+    #[cfg(feature = "eigenvalue")]
     #[test]
     fn test_estimate_occupancy_noise_only() {
         let config = FieldModelConfig {

--- a/rust-port/wifi-densepose-rs/crates/wifi-densepose-train/Cargo.toml
+++ b/rust-port/wifi-densepose-rs/crates/wifi-densepose-train/Cargo.toml
@@ -27,7 +27,7 @@ cuda = ["tch-backend"]
 
 [dependencies]
 # Internal crates
-wifi-densepose-signal = { version = "0.3.0", path = "../wifi-densepose-signal" }
+wifi-densepose-signal = { version = "0.3.0", path = "../wifi-densepose-signal", default-features = false }
 wifi-densepose-nn = { version = "0.3.0", path = "../wifi-densepose-nn" }
 
 # Core

